lib/app/url-handlers.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2025, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import url from 'node:url';
+import type {NextFunction, Request, Response} from 'express';
+
+import {logger} from '../logger.js';
+import {PropertyGetter} from '../properties.interfaces.js';
+import {ShortLinkResolver} from '../shortener/google.js';
+
+/**
+ * Detects if the request is from a mobile viewer
+ * @param req - Express request object
+ * @returns true if the request is from a mobile viewer
+ */
+export function isMobileViewer(req: Request): boolean {
+    return req.header('CloudFront-Is-Mobile-Viewer') === 'true';
+}
+
+/**
+ * Handles legacy Google URL shortener redirects
+ */
+export class OldGoogleUrlHandler {
```

Comment:
Maybe we can call it legacy? Perhaps something like this
```suggestion
export class LegacyGoogleUrlHandler {
```

Reply:
Done